core/src/main/java/com/netflix/conductor/core/execution/ParametersUtils.java

```diff
@@ -114,7 +116,7 @@ public ParametersUtils() {
                     taskParams.put("reasonForIncompletion", task.getReasonForIncompletion());
                     taskParams.put("callbackAfterSeconds", task.getCallbackAfterSeconds());
                     taskParams.put("workerId", task.getWorkerId());
-                    inputMap.put(task.getReferenceTaskName(), taskParams);
+                    inputMap.put(task.isLoopOverTask() ? DoWhileTaskMapper.getTaskDefReferenceName(task.getReferenceTaskName()) :  task.getReferenceTaskName(), taskParams);
```

Comment:
This is required to support input parameters in loopOver tasks as we are changing task refname in each iteration.

Reply:
similar leaky abstraction as the comment above. 

Can this static method be moved out of the TaskMapper class?